arduino/smartcar/smartcar.ino

```diff
@@ -28,6 +30,8 @@ const float STOPPING_SPEED = 0.3; //m/s. used to decide when to stop in slowDown
 const int PULL_OVER_DISTANCE = 250; //value used for connectivityLoss(), as how far the car pulls over
 bool isParked = true;
 
+std::vector<char> frameBuffer;
```

Comment:
Cool! So the vector stores each frame as characters? If so, what is the difference from a dynamic array? Just curious

Reply:
The vector is a dynamic array aka the same as ArrayList is a dynamic array in Java. It has the same logic, once the size is fully used it will resize it automatically. For the first question, I am not entirely sure but what I can say is camera flows pixel by pixel in a single image through MQTT with the values of r,g,b, and a. Those values would probably be the reason for using the char type.

Reply:
char is used as a "byte" type in this case as we just want to store bytes without other type info. Alternatives are uint8_t or  arduino specific [byte](https://www.arduino.cc/reference/en/language/variables/data-types/byte/); basically any type that is 8 bits in size will do for this case. this is in no way related to text or the values of the image.

std::vector is used so that you can change the size of the buffer at runtime (it allocates to the heap); an alternative would be to use a std::array but that requires you to know the size of the frame at compile time with added the benefit of not having to do an expensive memory allocation. basically static vs dynamic

If you look at how the vector is used:
`frameBuffer.resize(Camera.width() * Camera.height() * Camera.bytesPerPixel());`  
resizes the vector to perfectly fit the whole image, then:
`Camera.readFrame(frameBuffer.data());`  
copies all the bytes of the current frame into `frameBuffer`